addons/pos_discount/static/src/js/DiscountButton.js

```diff
@@ -18,7 +18,9 @@ odoo.define('pos_discount.DiscountButton', function(require) {
                 startingValue: this.env.pos.config.discount_pc,
             });
             if (confirmed) {
-                const val = Math.round(Math.max(0,Math.min(100,parseFloat(payload))));
+                const parsedPayload = parseFloat(payload);
+                const clampedValue = Math.max(0, Math.min(100, parsedPayload));
+                const val = Math.round(clampedValue * 10) / 10;
```

Comment:
```suggestion
                const val = Math.max(0,Math.min(100,parseFloat(payload)));
```
How about completely skipping the rounding? Because here, we are rounding to 1 decimal place, then later, somebody will come back to ask to accurately compute discount of 5.55%.